packages/fxa-settings/src/components/Settings/FlowRecoveryKeyHint/index.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState } from 'react';
+import { FlowContainer } from '../FlowContainer';
+import { ProgressBar } from '../ProgressBar';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { useForm } from 'react-hook-form';
+import InputText from '../../InputText';
+import { useFtlMsgResolver, useAccount } from '../../../models';
+import { logViewEvent, usePageViewEvent } from '../../../lib/metrics';
+import { LightbulbImage } from '../../images';
+
+export type FlowRecoveryKeyHintProps = {
+  navigateForward: () => void;
+  navigateBackward: () => void;
+  localizedPageTitle: string;
+};
+
+type HintForm = { hint: string };
+
+export const viewName = 'settings.account-recovery.hint';
+export const saveHintSuccessEventName = 'save-hint-success';
+export const saveHintFailureEventName = 'save-hint-failure';
+export const backwardNavigationEventName = 'navigate-backward';
+export const forwardNavigationEventName = 'navigate-forward';
+
+export const FlowRecoveryKeyHint = ({
+  navigateForward,
+  navigateBackward,
+  localizedPageTitle,
+}: FlowRecoveryKeyHintProps) => {
+  /*
+   * TODO:
+   * logging a page view event for each step of the flow may be excessive --
+   * resolve what metrics are needed and adjust accordingly in FXA-7249
+   */
+  const account = useAccount();
+  const [hintError, setHintError] = useState<string>();
+
+  const hintForm = useForm<HintForm>({
+    mode: 'onTouched',
+  });
+
+  const onSubmit = async ({ hint }: HintForm) => {
+    if (hint && hint.length > 0) {
```

Comment:
Looking at the typedef for HintForm above, it looks like `hint` is a string.
Q: Do we need to do something like `hint.trim().length` here to guard against somebody typing a space into the hint field, or is that handled elsewhere, or possibly not needed?